include/options/options.php

```diff
@@ -0,0 +1,506 @@
+		$errors->add(
+			'pll_unknown_option_keys',
+			sprintf(
+				/* translators: %s is a list of option names. */
+				_n( 'Unknown option key %s.', 'Unknown option keys %s.', count( $values ), 'polylang' ),
+				wp_sprintf_l( '%l', $this->wrap_in_code( array_keys( $values ) ) )
+			)
+		);
+
+		return $errors;
+	}
+
+	/**
+	 * Returns JSON schema for all options of the current blog.
+	 *
+	 * @since 3.7
+	 *
+	 * @return array The schema.
+	 */
+	public function get_schema(): array {
+		if ( isset( $this->schema[ $this->current_blog_id ] ) ) {
+			return $this->schema[ $this->current_blog_id ];
+		}
+
+		$properties = array();
+
+		if ( ! empty( $this->options[ $this->current_blog_id ] ) ) {
+			foreach ( $this->options[ $this->current_blog_id ] as $option ) {
+				if ( ! $option instanceof Abstract_Option ) {
+					continue;
+				}
+
+				$sub_schema = $option->get_schema();
+
+				// Cleanup.
+				unset( $sub_schema['title'], $sub_schema['$schema'] );
+
+				$properties[ $option->key() ] = $sub_schema;
+			}
+		}
+
+		$this->schema[ $this->current_blog_id ] = array(
+			'$schema'              => 'http://json-schema.org/draft-04/schema#',
+			'title'                => static::OPTION_NAME,
+			'description'          => __( 'Polylang options', 'polylang' ),
+			'type'                 => 'object',
+			'context'              => array( 'edit' ),
+			'properties'           => $properties,
+			'additionalProperties' => false,
+		);
+
+		return $this->schema[ $this->current_blog_id ];
+	}
+
+	/**
+	 * Tells if an option exists.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key The name of the option to check for.
+	 * @return bool
+	 */
+	public function has( string $key ): bool {
+		return isset( $this->options[ $this->current_blog_id ][ $key ] ) && $this->options[ $this->current_blog_id ][ $key ] instanceof Abstract_Option;
+	}
+
+	/**
+	 * Returns the value of the specified option.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key The name of the option to retrieve.
+	 * @return mixed
+	 */
+	public function &get( string $key ) {
+		if ( ! $this->has( $key ) ) {
+			return null;
+		}
+
+		/** @phpstan-var Abstract_Option */
+		$option = $this->options[ $this->current_blog_id ][ $key ];
+		$value  = $option->get();
+		return $value;
+	}
+
+	/**
+	 * Assigns a value to the specified option.
+	 * This doesn't allow to set an unknown option.
+	 * When doing multiple `set()`, options must be set in the right order: some options depend on other options' value.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $key   The name of the option to assign the value to.
+	 * @param mixed  $value The value to set.
+	 * @return WP_Error
+	 */
+	public function set( string $key, $value ): WP_Error {
+		if ( ! $this->has( $key ) ) {
+			/* translators: %s is the name of an option. */
+			return new WP_Error( 'pll_unknown_option_key', sprintf( __( 'Unknown option key %s.', 'polylang' ), $this->wrap_in_code( $key ) ) );
```

Comment:
Why do we need this error message?

Reply:
To know why the option has not been set.